Modification to R environment

glennpj opened the pull request:
This PR modifies how the R environmnet is presented, and fixes
installing the standalone Rmath library.

- The Rmath build and install methods are combined into one
- Set parallel=False when installing Rmath
- remove the run environment that set up variables for libraries and
  headers that are not really needed, and pollute the environment.

glennpj committed:
Modification to R environment

This PR modifies how the R environmnet is presented, and fixes
installing the standalone Rmath library.

- The Rmath build and install methods are combined into one
- Set parallel=False when installing Rmath
- remove the run environment that set up variables for libraries and
  headers that are not really needed, and pollute the environment.

alalazo commented:
It seems the function removed here is used in the documentation in:
```
/home/runner/work/spack/spack/lib/spack/docs/module_file_support.rst:140
```

glennpj commented:
> It seems the function removed here is used in the documentation in:
> 
> ```
> /home/runner/work/spack/spack/lib/spack/docs/module_file_support.rst:140
> ```

Oh, I forgot that was there. While none of those variables are needed to build R packages outside of Spack, LD_LIBRARY_PATH might still be useful if the `+rmath` variant is used. I will rework this a bit and modify the docs.

glennpj committed:
Add setup_run_environment back

- Add back the setup_run_environment with LD_LIBRARY_PATH and
  PKG_CONFIG_PATH.
- Adjust documentation to reflect the current code.
